lib/guardian.rb

```diff
@@ -620,10 +620,14 @@ def is_me?(other)
   private
 
   def is_my_own?(obj)
-    unless anonymous?
-      return obj.user_id == @user.id if obj.respond_to?(:user_id) && obj.user_id && @user.id
-      return obj.user == @user if obj.respond_to?(:user)
+    if anonymous?
+      return(
+        SiteSetting.allow_anonymous_likes? && obj.class == PostAction && obj.is_like? &&
+          obj.user_id == @user.id
+      )
```

Comment:
stree really wanted this. :/ 

Originally this looked like:
`return SiteSetting.allow_anonymous_likes? && obj.class == PostAction && obj.is_like? && obj.user_id == @user.id if if anonymous?`